pyccel/ast/numpyext.py

```diff
@@ -1322,6 +1323,17 @@ class NumpyArctanh(NumpyUfuncUnary):
 
 #=======================================================================================
 
+
+class NumpySign(NumpyUfuncUnary):
+    """Represent a call to the sign function in the Numpy library"""
+    __slots__ = ()
+    name = 'sign'
+    def _set_dtype_precision(self, x):
+        if not isinstance(x.dtype, (NativeInteger, NativeFloat, NativeComplex)):
+            raise TypeError(f'{x.dtype} not supported')
+        self._dtype     = x.dtype
+        self._precision = get_final_precision(x)
```

Comment:
Is this the expected behaviour? Windows doesn't seem to like it

Reply:
Yeah, for some reason the generated python array does not have the same dtype as pyccel's. I'm working on it.

Reply:
I thought maybe I was using `long int isign` instead of `long long int isign` but it doesn't seem to be the issue.

Reply:
It's a good job we modified those tests ;)

Given it's only appearing on Windows it's almost certainly got something to do with how int is handled. Int is int32 by default on Windows and int64 everywhere else but some functions like to do things differently. I'm guessing this line is linked to the problem anyway.

Reply:
It's odd that only the arrays are complaining though